pkg/controller/infinispan/resources/secret/secret_controller.go

```diff
@@ -124,6 +138,118 @@ func (s secretResource) createSecret(name, label string, identities []byte) erro
 	return nil
 }
 
+func (s secretResource) reconcileKeystoreSecret() (*reconcile.Result, error) {
+	i := s.infinispan
+	if !i.IsEncryptionEnabled() {
+		return nil, nil
+	}
+
+	keystoreSecret := &corev1.Secret{}
+	if result, err := kube.LookupResource(i.GetKeystoreSecretName(), i.Namespace, keystoreSecret, s.client, s.log); result != nil {
+		return result, err
+	}
+
+	_, err := controllerutil.CreateOrUpdate(context.TODO(), s.client, keystoreSecret, func() error {
+		return controllerutil.SetControllerReference(i, keystoreSecret, s.scheme)
+	})
+	if err != nil {
+		return &reconcile.Result{}, err
+	}
+	return nil, nil
+}
+
+func (s secretResource) reconcileTruststoreSecret() (*reconcile.Result, error) {
+	i := s.infinispan
+	if !i.IsClientCertEnabled() {
+		return nil, nil
+	}
+
+	trustSecretName := s.infinispan.GetTruststoreSecretName()
+	generateTruststore := func(caPem []byte, data map[string][]byte) error {
+		if _, passwordProvided := data[consts.EncryptTruststorePasswordKey]; !passwordProvided {
+			data[consts.EncryptTruststorePasswordKey] = []byte(EncryptTruststorePassword)
+		}
+
+		certs := [][]byte{caPem}
+		for certKey := range data {
+			if strings.HasPrefix(certKey, EncryptClientCertPrefix) {
+				certs = append(certs, data[certKey])
+			}
+		}
+		password := string(data[consts.EncryptTruststorePasswordKey])
+		truststore, err := security.GenerateTruststore(certs, password)
+		if err != nil {
+			return err
+		}
+		data[consts.EncryptTruststoreKey] = truststore
+		return nil
+	}
+
+	trustSecret := &corev1.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      trustSecretName,
+			Namespace: i.Namespace,
+		},
+	}
+
+	if i.IsEncryptionCertFromService() {
+		// Get secret if it exists
+		// If it exists, create ts with client certs
+		// Else create secret and just use service-ca.crt
+		operatorTokenSecret, err := kube.LookupOperatorTokenSecret(s.client, s.log)
+		if err != nil {
+			return &reconcile.Result{}, err
+		}
+		caPem := operatorTokenSecret.Data["service-ca.crt"]
+
+		_, err = kube.CreateOrPatch(ctx, s.client, trustSecret, func() error {
+			if err := generateTruststore(caPem, trustSecret.Data); err != nil {
+				return err
+			}
+			return controllerutil.SetControllerReference(i, trustSecret, s.scheme)
```

Comment:
@ryanemerson wdyt about to add protection from the Secret name sharing between Infinispan CR as this is customer provided name and could be utilized in separate CR.
`metav1.IsControlledBy` could help us there

Reply:
`IsControlledBy` requires the controlling object, so I don't think we can use that as we may not have access or knowledge of the object. I think we can use `metav1.GetControllerOf` instead. If it returns a non-nil value, then output a error informing the user that the secret cannot be shared between CRs.

Reply:
Yeah, exactly  as we don't know controlled object